idea/src/org/jetbrains/kotlin/idea/inspections/RedundantSemicolonInspection.kt

```diff
@@ -57,6 +58,16 @@ class RedundantSemicolonInspection : AbstractKotlinInspection(), CleanupLocalIns
 
         if (semicolon.parent is KtEnumEntry) return false
 
+        (semicolon.parent.parent as? KtClass)?.let {
+            if (it.isEnum() && it.getChildrenOfType<KtEnumEntry>().isEmpty()) {
+                if (semicolon.prevLeaf { it !is PsiWhiteSpace && it !is PsiComment && !it.isLineBreak() }?.node?.elementType == KtTokens.LBRACE &&
+                    semicolon.nextLeaf { it !is PsiWhiteSpace && it !is PsiComment && !it.isLineBreak() }?.node?.elementType != KtTokens.RBRACE) {
```

Comment:
I thinks it's more precise just to check here that `getDeclarations()` isn't empty, instead of doing these operations with `nextLeaf`.